xml/upgrade-3.10-0.xsl

```diff
@@ -18,10 +23,74 @@
 <xsl:strip-space elements="*"/>
 <xsl:output encoding="UTF-8" indent="yes" omit-xml-declaration="yes"/>
 
-<!-- Copy everything unaltered by default -->
+<!--
+ Copy everything unaltered by default, except sort nvset elements by score.
+
+ nvset elements include the following:
+ * cluster_property_set
+ * instance_attributes
+ * meta_attributes
+ * utilization
+
+ This doesn't affect behavior. nvset elements of a given type within a given
+ parent element are processed in order of their score attributes, with a
+ nonexistent score treated as 0. In the event of a tie, elements are processed
+ in document order. The sort here preserves document order in the event of a
+ tie.
+
+ The order of nvset elements relative to non-nvset elements does not matter. So
+ this template puts existing nvset elements after existing non-nvset elements.
+
+ This facilitates later transformations, allowing us to more easily drop nvpairs
+ with unsupported values without changing behavior.
+ -->
 <xsl:template match="/|@*|node()">
     <xsl:copy>
-        <xsl:apply-templates select="@*|node()"/>
+        <xsl:variable name="nvsets"
+                      select="cluster_property_set
+                              |instance_attributes
+                              |meta_attributes
+                              |utilization"/>
+
+        <!-- XPath 1.0 set difference idiom -->
+        <xsl:variable name="non_nvsets"
+                      select="@*|node()[count(.|$nvsets) != count($nvsets)]"/>
+
+        <xsl:apply-templates select="$non_nvsets"/>
+        <xsl:apply-templates select="$nvsets">
```

Comment:
I just noticed that nvsets can also be id-refs -_- So the sort needs to look up the id and score from the referenced nvset in that case.